pkg/controller/flow/reconcile.go

```diff
@@ -0,0 +1,360 @@
+		return reconcile.Result{}, err
+	}
+
+	// Requeue if the resource is not ready:
+	return reconcile.Result{Requeue: !flow.Status.IsReady()}, err
+}
+
+func (r *reconciler) reconcile(flow *v1alpha1.Flow) error {
+	// See if the flow has been deleted
+	accessor, err := meta.Accessor(flow)
+	if err != nil {
+		log.Fatalf("Failed to get metadata: %s", err)
+	}
+	deletionTimestamp := accessor.GetDeletionTimestamp()
+	glog.Infof("DeletionTimestamp: %v", deletionTimestamp)
+
+	target, err := r.resolveActionTarget(flow.Namespace, flow.Spec.Action)
+	if err != nil {
+		glog.Warningf("Failed to resolve target %v : %v", flow.Spec.Action, err)
+		flow.Status.PropagateActionTargetResolved(corev1.ConditionFalse, "ActionTargetNotResolved", err.Error())
+		return err
+	}
+
+	flow.Status.PropagateActionTargetResolved(corev1.ConditionTrue, "ActionTargetResolved", fmt.Sprintf("Resolved to: %q", target))
+
+	// Ok, so target is the underlying k8s service (or URI if so specified) that we want to target
+	glog.Infof("Resolved Target to: %q", target)
+
+	// Reconcile the Channel. Creates a channel that is the target that the Feed will use.
+	// TODO: We should reuse channels possibly.
+	channel, err := r.reconcileChannel(flow)
+	if err != nil {
+		glog.Warningf("Failed to reconcile channel : %v", err)
+		return err
+	}
+	flow.Status.PropagateChannelStatus(channel.Status)
+
+	subscription, err := r.reconcileSubscription(channel.Name, target, flow)
+	if err != nil {
+		glog.Warningf("Failed to reconcile subscription : %v", err)
+		return err
+	}
+	flow.Status.PropagateSubscriptionStatus(subscription.Status)
+
+	channelDNS := channel.Status.DomainInternal
+	if channelDNS != "" {
+		glog.Infof("Reconciling feed for flow %q targeting %q", flow.Name, channelDNS)
+		feed, err := r.reconcileFeed(channelDNS, flow)
+		if err != nil {
+			glog.Warningf("Failed to reconcile feed: %v", err)
+			return err
+		}
+		glog.Infof("Reconciled feed status for flow %q targeting %q : %+v", flow.Name, channelDNS, feed.Status)
+		flow.Status.PropagateFeedStatus(feed.Status)
+	}
+	return nil
+}
+
+func (r *reconciler) updateStatus(flow *v1alpha1.Flow) (*v1alpha1.Flow, error) {
+	newFlow := &v1alpha1.Flow{}
+	err := r.client.Get(context.TODO(), client.ObjectKey{Namespace: flow.Namespace, Name: flow.Name}, newFlow)
+
+	if err != nil {
+		return nil, err
+	}
+	newFlow.Status = flow.Status
+
+	// Until #38113 is merged, we must use Update instead of UpdateStatus to
+	// update the Status block of the Flow resource. UpdateStatus will not
+	// allow changes to the Spec of the resource, which is ideal for ensuring
+	// nothing other than resource status has been updated.
+	if err = r.client.Update(context.TODO(), newFlow); err != nil {
+		return nil, err
+	}
+	return newFlow, nil
+}
+
+// syncHandler compares the actual state with the desired, and attempts to
+// converge the two. It then updates the Status block of the Flow resource
+// with the current status of the resource.
```

Comment:
This comment is about a different function :smile:

Reply:
done